annotate.el

```diff
@@ -1400,26 +1410,100 @@ The searched interval can be customized setting the variable:
           (concat " \n" (make-string annotate-annotation-column ? ))
         (make-string prefix-length ? )))))
 
+(defun annotate-annotations-at (pos)
+  "Returns the annotations (overlay where (annotationp overlay) -> t)
+   at positions pos or nil if no annotations exists at pos"
+  (cl-remove-if-not #'annotationp
+                    (overlays-at pos)))
+
+(defun annotate-previous-annotation-ends (pos)
+  "Returns the previous annotation that ends before pos or nil if no annotation
+was found.
+NOTE this assumes that annotations never overlaps"
+  (cl-labels ((previous-annotation-ends (start)
+                (let ((all-annotations (annotate-annotations-at start)))
+                  (while (and (>= (1- start)
+                                  (point-min))
+                              (null all-annotations))
+                    (setf start (1- start))
+                    (setf all-annotations (annotate-annotations-at (1- start))))
+                  all-annotations)))
+    (let ((all-annotations (annotate-annotations-at pos)))
+      (if all-annotations
+          (previous-annotation-ends (1- (overlay-start (cl-first all-annotations))))
```

Comment:
I think `all-annotations` should be sorted by `overlay-start` before `cl-first`, right?